### PR TITLE
Refactor some PaintSession things

### PR DESCRIPTION
**Pull request by ZehMatt**
This gains me +10 fps for EverythingPark.park at 1080p at its saved zoom/rotation, it went from 30 to 40 FPS with the Software (Hardware) renderer. The PR puts all the memory closer together, PaintSession will have 1024 paint entry slots and will only use heap once that's full, PaintSession is now stored in a segmented vector rather than using make_unique which just allocates memory wherever something is currently free. There is also a small reduction in memory usage, 602 down to 588 MiB having EverythingPark loaded at its saved rotation/zoom.

### CODE REVIEW

**Review by AaronVanGeffen — APPROVED** [review]
Code-wise, I like how this PR swaps out custom data structures for more `sfl` ones. This obviously removes some of the maintenance burden on our end. Moreover, with how well-tested the `sfl` data structures are, I think we can safely say the code's robustness is improved as well.

On my end, I'm not seeing a clear performance improvement, nor any regressions, though. Unfortunately, we don't have a standard benchmark for this either. Especially with entities moving around, it's hard to have a consistent long-term measure. Regardless, I'm happy to approve this PR.

File: src/openrct2/paint/Painter.cpp [hunk 1]
```diff
@@ -197,15 +195,13 @@ void Painter::ReleaseSession(PaintSession* session)
...
 }
 
 Painter::~Painter()
 {
-    for (auto&& session : _paintSessionPool)
-    {
-        ReleaseSession(session.get());
-    }
+    _paintSessionPool.clear();
     _paintSessionPool.clear();
```
Comment (Gymnasiast) [inline src/openrct2/paint/Painter.cpp:206]: This calls the same function twice, is that intentional?

Comment (ZehMatt) [reply src/openrct2/paint/Painter.cpp:206]: Good catch, don't know how this came to be.

**Review by janisozaur — APPROVED** [review]
I haven't done all my tests just yet, I will try to continue working with that once this PR is merged. Overall, I haven't found anything here to complain about and it seems to improve things.

**Pull request merged**